client/gatsby-browser.js

```diff
@@ -27,3 +28,11 @@ wrapRootElement.propTypes = {
 export const wrapPageElement = layoutSelector;
 
 export const disableCorePrefetching = () => true;
+
+export const onClientEntry = () => {
+  // purge the csrf cookies, rather than relying what the browser decides a
+  // Session duration is
+  cookies.erase('_csrf');
+  // the token must be erased since it is only valid for the old _csrf secret
+  cookies.erase('csrf_token');
+};
```

Comment:
So '_csrf' abd csrf_token' are regenerated on every refresh. I was just wondering if that in intended.

Reply:
It is, yes. If `_csrf` is not manually regenerated it can persist indefinitely. Since we're regenerating that we have to force the token to be regenerated or it will be invalid.